st2actions/st2actions/query/mistral/v2.py

```diff
@@ -101,6 +101,11 @@ def _get_workflow_tasks(self, exec_id):
         resp = requests.get(url)
         result = resp.json()
         tasks = result.get('tasks', [])
+
+        for task in tasks:
+            for attr in ['result', 'input', 'output']:
+                task[attr] = jsonify.try_loads(task.get(attr, None))
```

Comment:
Pretty much the reason we moved away from wsme ourselves.

>  It's not always the case we have control over how other API are written.

In this case we do :). It will definitely be beneficial to mistral to store this as json. If it were an opaque system then of course the question does not arise and we deal with it wherever possible.

Looks like you've already checked in the code so fair to assume you will not bother exploring a fix in mistral?